yamllint/rules/quoted_strings.py

```diff
@@ -52,27 +52,32 @@
 def check(conf, token, prev, next, nextnext, context):
     quote_type = conf['quote-type']
 
-    if (isinstance(token, yaml.tokens.ScalarToken) and
-            isinstance(prev, (yaml.ValueToken, yaml.TagToken))):
-        # Ignore explicit types, e.g. !!str testtest or !!int 42
-        if (prev and isinstance(prev, yaml.tokens.TagToken) and
-                prev.value[0] == '!!'):
-            return
+    if not isinstance(token, yaml.tokens.ScalarToken):
+        return
+
+    if not isinstance(prev, (yaml.ValueToken, yaml.TagToken)):
+        return
```

Comment:
For readability, could you simplify to:
```suggestion
    if not (isinstance(token, yaml.tokens.ScalarToken) and
            isinstance(prev, (yaml.ValueToken, yaml.TagToken))):
```